app/src/main/java/com/example/pinterbest/ProfileFragment.kt

```diff
@@ -19,6 +35,12 @@ class ProfileFragment : Fragment() {
     private var _backgroundProfile: FrameLayout? = null
     private val backgroundProfile get() = _backgroundProfile!!
 
+    var avatarLink: String? = null
+    var followers: Int? = null
+    var following: Int? = null
+
+    private val apiService = ApiClient().retrofit.create(ApiService::class.java)
```

Comment:
you are creating ApiService instance in few places. Let's make it singleton

Reply:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/3/commits/f47063b96f0a24a56ad6def81896bbe4dee63167